controlplane/kubeadm/main.go

```diff
@@ -158,7 +159,8 @@ func main() {
 	if profilerAddress != "" {
 		klog.Infof("Profiler listening for requests at %s", profilerAddress)
 		go func() {
-			klog.Info(http.ListenAndServe(profilerAddress, nil))
+			srv := http.Server{Addr: profilerAddress, ReadHeaderTimeout: 2 * time.Second}
```

Comment:
This was done to address a new rule in gosec (G112) to prevent a slow loris attack. I'm happy to alternatively ignore that here as this is for debug purposes.